src/ucs/config/parser.c

```diff
@@ -1851,6 +1852,69 @@ void ucs_config_parser_print_all_opts(FILE *stream, const char *prefix,
     }
 }
 
+static void ucs_config_parser_search_similar_variables(
+        const ucs_config_field_t *config_table, const char *env_prefix,
+        const char *table_prefix, const char *unused_var,
+        ucs_string_buffer_t *matches_buffer)
+{
+    char ucx_var[128]                 = {0};
+    const uint32_t MAX_FUZZY_DISTANCE = 3;
+    const ucs_config_field_t *field;
+    int distance;
+
+    for (field = config_table; !ucs_config_field_is_last(field); ++field) {
+        if (ucs_config_is_table_field(field)) {
+            ucs_config_parser_search_similar_variables(field->parser.arg,
+                                                       env_prefix, table_prefix,
+                                                       unused_var,
+                                                       matches_buffer);
+        }
+        else {
+            snprintf(ucx_var, sizeof(ucx_var), "%s%s%s", env_prefix,
+                     table_prefix ? table_prefix : "", field->name);
+
+            distance = ucs_string_distance(unused_var, ucx_var);
+            if (distance < 0) {
+                ucs_warn("Failed to calculate string distance: max string "
+                         "length exceeded: unused_var: %lu, ucx_var %lu",
+                          strlen(unused_var), strlen(ucx_var));
+            }
```

Comment:
i think we can just ignore (or ucs_debug) failures in this call. it's not like the user has something to do about it